ui/src/message_popup/renderers/minimal_timed_view.jsx

```diff
@@ -0,0 +1,88 @@
+/* @flow weak */
+import React from 'react';
+import RaisedButton from 'material-ui/RaisedButton';
+
+var sceneTimer;
+
+// This response supports responding with text.
+export default React.createClass({
+  displayName: 'MinimalTimedView',
+
+  propTypes: {
+    onLogMessage: React.PropTypes.func.isRequired,
+    onResponseSubmitted: React.PropTypes.func.isRequired,
+    recordText: React.PropTypes.string,
+  },
+
+  getDefaultProps() {
+    return {
+      recordText: 'Respond',
+      autoFocus: true,
+      textHeight: 48
+    };
+  },
+
+
+  getInitialState() {
+    sceneTimer = setTimeout(this.onTimer, 25000);
+    return {
+      responseText: 'type here!'
+    };
+  },
+
+  onTextChanged(e) {
+    const responseText = e.target.value;
+    this.setState({responseText});
+  },
+
+  onTimer() {
+    const {responseText} = this.state;
+    this.props.onLogMessage('message_popup_text_response', {responseText});
+    this.props.onResponseSubmitted({ignore: true});
+  },
+
+
+  onSkipTapped() {
+    clearTimeout(sceneTimer);
+    const {responseText} = this.state;
+    this.props.onLogMessage('message_popup_text_response', {responseText});
```

Comment:
We should update the message type to `timer_auto_advance_response` or something similar (same as above).  Also, we probably want to capture in the data when people are skipping versus getting forced along.  So here you could add some data in the second argument:

```
this.props.onLogMessage('timer_auto_advance_response', {skipTapped: true});
```

That way we can tell looking in the data on the server if we want to know how often people are skipped versus getting forced along.